tools/gen_irr_diag.c

```diff
@@ -266,18 +271,18 @@ int irr_diag_scalar_indices( char *dirname )
    sprintf( line,"  chm_opts_ndx(:nchm_opts) = (/ ");
    for( i = 0; i < nChmOpts; i++ ) {
      if( i == 0 ) 
-       sprintf( piece,"%s_kpp",chm_scheme[i]);
+       snprintf( piece,135,"%.128s_kpp",chm_scheme[i]);
      else
-       sprintf( piece," ,%s_kpp",chm_scheme[i]);
+       snprintf( piece,135," ,%.128s_kpp",chm_scheme[i]);
      strcat( line,piece );
    }
    strcat( line," /)\n" );
    fprintf( fp_inc,line );
    fprintf( fp_inc," \n");
 
-   for( i = 0; i < nChmOpts,rxt_cnt[i] > 0; i++ ) {
+   for( i = 0; /*i < nChmOpts &&*/ rxt_cnt[i] > 0; i++ ) {
```

Comment:
It seems like this is relying on `rxt_cnt` to be default initialized to zero. I'm not too sure what it is used for, but given that all other loops use `nChmOpts` to specify number of indices into `rxt_cnt`, keeping this check seems prudent

Reply:
The old version was ignoring the result of the `nChmOpts` comparison. Are you suggesting that check be included in the for-loop exit condition?

Reply:
Ah! You're correct, I forgot that this use of the `,` operator is nulling and not `&&`

I'd imagine that the original intent was to have it included via `&&` given the other loops. Maybe @weiwangncar or someone else with historic knowledge on this section of the Registry can chime in. Otherwise, as long as tests agree, I think that check _should_ be there

Reply:
Changed to the suggested form.  I'd still like to get an opinion from someone who actually knows this bit of the code.